mlx/transforms.cpp

```diff
@@ -120,7 +146,6 @@ void simplify(const std::vector<array>& outputs) {
   while (!tape.empty()) {
     auto arr = std::move(tape.front());
     tape.pop();
-
     if (cache.find(arr.id()) != cache.end()) {
```

Comment:
> We don't seem to be adding anything to the cache.

Can you explain what you mean? I did not remove the line which adds to the cache.. or did you check and see nothing was getting added?

Reply:
Yeah you didn't remove it. I don't know when it got removed :-) . Basically after using the `cache` to do the DFS then we clear it and then we never add anything to it again.

Now my comment doesn't make sense cause we shouldn't be checking the cache during the tape loop at all so the thing to be removed really is the `if (cache.find(arr.id()) != cache.end())`. Since things will be put to the tape only once it isn't possible to have an array twice in the tape anyway. No need for that check.